ppafm/common.py

```diff
@@ -436,90 +443,86 @@ def maxAlongDirEntropy(atoms, hdir, beta=1.0):
 # ==============================
 
 
-def autoGridN():
-    params["gridN"][0] = round(np.linalg.norm(params["gridA"]) * 10)
-    params["gridN"][1] = round(np.linalg.norm(params["gridB"]) * 10)
-    params["gridN"][2] = round(np.linalg.norm(params["gridC"]) * 10)
-    return params["gridN"]
+def autoGridN(parameters):
+    parameters.gridN[0] = round(np.linalg.norm(parameters.gridA) * 10)
+    parameters.gridN[1] = round(np.linalg.norm(parameters.gridB) * 10)
+    parameters.gridN[2] = round(np.linalg.norm(parameters.gridC) * 10)
+    return parameters.gridN
 
 
 # overide default parameters by parameters read from a file
-def loadParams(fname):
+def loadParams(fname, parameters):
```

Comment:
Would this function be a better fit as a method of the `PpafmParameters` class? It seems like we always use this function right after constructing the parameters object, so it could even be made into a class method like:
```python
class PpafmParameters(pydantic.BaseModel):
    ...
    @classmethod
    def from_file(cls, file_path="./params.ini"):
        parameters = cls()
        # populate parameters....
        return parameters

parameters = PpafmParameters.from_file("./params.ini")
```

Reply:
done in https://github.com/Probe-Particle/ppafm/pull/275/commits/1f1e6f8f92e40edaf76fe3d3bd278f1dfaf393d3